core_backend/app/routers/manage_content.py

```diff
@@ -51,21 +71,33 @@ async def edit_content(
     Edit content endpoint
     """
     old_content = await get_content_from_db(
-        content_id,
+        content_text_id,
         asession,
     )
 
     if not old_content:
         raise HTTPException(
-            status_code=404, detail=f"Content id `{content_id}` not found"
+            status_code=404, detail=f"Content text id `{content_text_id}` not found"
         )
-    updated_content = await update_content_in_db(
-        content_id,
-        content,
-        asession,
+
+    is_valid, reason = await validate_edit_content(
+        content=content,
+        asession=asession,
+        old_content=old_content,
     )
+    if is_valid:
```

Comment:
If I update an existing content text by associating it with a different content_id, then we can be left with a content_id in the content table that has no content text associated.

After updating, we can clean up dangling contents if they exist?

Reply:
I will add that on update as well. Thanks for catching that.